### PR TITLE
Add OD to Curve/ revise lat/long

### PR DESCRIPTION
**Pull request by UrbanGISer**
Add OD to Curve node and revise the lat/long to lat/lon

### CODE REVIEW

File: knime_extension/src/nodes/transform.py [hunk 1]
```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
...
+
+    num_points = knext.IntParameter(
+        "Number of points ",
+        "Specify the number of points to define the Bézier curve.",
+        default_value=100,
+        is_advanced=True,
+    )
+
+    height_scale = knext.DoubleParameter(
+        "Hight scale ",
```
Comment (koettert) [inline knime_extension/src/nodes/transform.py:693]: Height

File: knime_extension/src/nodes/transform.py [hunk 2]
```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
...
+        "Set the scale factor for the curve's height, controlling the distance of the curve from the straight line.",
+        default_value=0.3,
+        is_advanced=True,
+    )
+
+    angle_degrees = knext.IntParameter(
+        "Curve angle (degrees) ",
+        "Define the angle that controls the shape and direction of the Bézier curve.",
+        default_value=60,
+        is_advanced=True,
```
Comment (koettert) [inline knime_extension/src/nodes/transform.py:703]: Max value 360 and min value 0

File: knime_extension/src/nodes/transform.py [hunk 3]
```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
...
+        "Specify the number of points to define the Bézier curve.",
+        default_value=100,
+        is_advanced=True,
+    )
+
+    height_scale = knext.DoubleParameter(
+        "Hight scale ",
+        "Set the scale factor for the curve's height, controlling the distance of the curve from the straight line.",
+        default_value=0.3,
+        is_advanced=True,
```
Comment (koettert) [inline knime_extension/src/nodes/transform.py:696]: Max value 1 and min value 0?

File: knime_extension/src/nodes/transform.py [hunk 4]
```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
...
+
+    d_geo_col = knext.ColumnParameter(
+        "Destination geopoint column",
+        "Select the geometry column that describes the destination.",
+        port_index=0,
+        column_filter=knut.is_geo_point,
+        include_row_key=False,
+        include_none_column=False,
+    )
+    result_settings = knut.ResultSettings(
```
Comment (koettert) [inline knime_extension/src/nodes/transform.py:680]: Why do you use the ResultSettings class here? This class was intended for single input column geometry nodes.

File: knime_extension/src/nodes/transform.py [hunk 5]
```diff
@@ -623,3 +623,168 @@ def execute(self, exec_context: knext.ExecutionContext, input_1):
...
+
+            # Calculate the points on the Bezier curve
+            curve_points = bezier_curve(p0, cp, p2, num_points=self.num_points)
+
+            # Convert the curve points to a LineString geometry
+            line_geom = LineString(curve_points)
+
+            return line_geom
+
+        if self.result_settings.mode == knut.ResultSettingsMode.APPEND.name:
```
Comment (koettert) [inline knime_extension/src/nodes/transform.py:781]: I would simply always append a new column to the input table with name Curve. Just make sure that the name is unique by using the `knut.get_unique_column_name()` method

**Review by koettert — CHANGES_REQUESTED** [review]

File: knime_extension/src/nodes/transform.py (hunk 1 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/transform.py]: revised

File: knime_extension/src/nodes/transform.py (hunk 2 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/transform.py:709]: added

File: knime_extension/src/nodes/transform.py (hunk 3 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/transform.py:700]: added

File: knime_extension/src/nodes/transform.py (hunk 4 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/transform.py]: revise to appending column

File: knime_extension/src/nodes/transform.py (hunk 5 above)
Comment (UrbanGISer) [reply knime_extension/src/nodes/transform.py]: revised

**Pull request merged**